seacrowd/sea_datasets/tcope/tcope.py

```diff
@@ -0,0 +1,162 @@
+from seacrowd.utils.constants import Licenses, Tasks
+
+_CITATION = """
+@article{gonzales_broadening_2023,
+  author    = {Gonzales, Wilkinson Daniel Wong},
+  title     = {Broadening horizons in the diachronic and sociolinguisstic study of
+  Philippine Englishes with the Twitter Corpus of Philippine Englishes (TCOPE)},
+  journal   = {English World-Wide},
+  year      = {2023},
+  url       = {https://osf.io/k3qzx},
+  doi       = {10.17605/OSF.IO/3Q5PW},
+}
+"""
+
+_LOCAL = False
+_LANGUAGES = ["eng", "fil"]
+_DATASETNAME = "tcope"
+_DESCRIPTION = """
+The TCOPE dataset consists of 1,048,576 public tweets (amounting to about 13.5 million words) collected from 13 major cities from the Philippines.
+Tweets are either purely in English or involve code-switching between English and Filipino.
+Tweets are tagged for part-of-speech and dependency parsing using spaCy. Tweets collected are from 2010 to 2021.
+The publicly available dataset is only a random sample (10%) from the whole TCOPE dataset, which consist of roughly 27 million tweets
+(amounting to about 135 million words) collected from 29 major cities during the same date range.
+"""
+
+_HOMEPAGE = "https://osf.io/3q5pw/wiki/home/"
+_LICENSE = Licenses.CC0_1_0.value
+_URL = "https://files.osf.io/v1/resources/3q5pw/providers/osfstorage/63737a5b0e715d3616a998f7"
+
+_SUPPORTED_TASKS = [Tasks.POS_TAGGING]
+_SOURCE_VERSION = "1.0.0"
+_SEACROWD_VERSION = "1.0.0"
+
+
+class TCOPEDataset(datasets.GeneratorBasedBuilder):
+    """TCOPE is a dataset of Philippine English tweets by Gonzales (2023)."""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    # Actual data has invalid "labels" likely due to coding errors,
+    # such as "BODY", "BIRTHDAY", "HAVAIANAS", etc. Only valid
+    # POS tags are included here and in loaded data.
+    pos_labels = ["NOUN", "PUNCT", "PROPN", "VERB", "PRON", "ADP", "ADJ", "ADV", "DET", "AUX", "PART", "CCONJ", "INTJ", "SPACE", "SCONJ", "NUM", "X", "SYM"]
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_source",
+            version=SOURCE_VERSION,
+            description=f"{_DATASETNAME} source schema",
+            schema="source",
+            subset_id=_DATASETNAME,
+        ),
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_seacrowd_seq_label",
+            version=SEACROWD_VERSION,
+            description=f"{_DATASETNAME} SEACrowd sequence labeling schema",
+            schema="seacrowd_seq_label",
+            subset_id=_DATASETNAME,
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "tcope_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "copeid": datasets.Value("string"),
+                    "userid": datasets.Value("int64"),
+                    "divided.tweet": datasets.Value("string"),
```

Comment:
suggestion: although it can differ from the source col naming, the ideal col naming should contain alphanumeric and underscore only w/o spaces